### ISSUE TITLE
Access violation on gifs with bad subimage data

### ISSUE THREAD

**Issue by strazzere** [kohler/gifsicle#106]
Similar to the others, likely unexploitable;

```
gdb$ run
Starting program: /home/tstrazzere/repo/gifsicle/src/gifsicle -O2 ./subimages.gif --output /dev/null
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
gifsicle:./subimages.gif:#0: read error: image corrupted, min_code_size too big
gifsicle:./subimages.gif:#0: read error: image corrupted, code out of range (19 times)
gifsicle:./subimages.gif:#0: read error: (not reporting more errors)
gifsicle:./subimages.gif:#0: read error: missing 6528 pixels of image data
gifsicle:./subimages.gif:#1: read error: image corrupted, min_code_size too big
gifsicle:./subimages.gif:#1: read error: missing 1344 pixels of image data
gifsicle:./subimages.gif:#2: read error: image corrupted, min_code_size too big
gifsicle:./subimages.gif:#2: read error: image corrupted, code out of range (19 times)
gifsicle:./subimages.gif:#2: read error: (not reporting more errors)
gifsicle:./subimages.gif:#2: read error: missing 1212 pixels of image data
gifsicle: (plus more errors; is this GIF corrupt?)
gifsicle: warning: too many colors, using local colormaps
  (You may want to try ‘--colors 256’.)
gifsicle:./subimages.gif: warning: some colors undefined by colormap
gifsicle:./subimages.gif: warning: some colors undefined by colormap
gifsicle:./subimages.gif: warning: some colors undefined by colormap
gifsicle:./subimages.gif: warning: some colors undefined by colormap

Program received signal SIGSEGV, Segmentation fault.
-----------------------------------------------------------------------------------------------------------------------[regs]
  RAX: 0x00000000006353C0  RBX: 0x020000000063F020  RBP: 0x0000000000004610  RSP: 0x00007FFFFFFFE110  o d I t s z a p c 
  RDI: 0x00007FFFF78B8620  RSI: 0x0000000000000330  RDX: 0x00000000006353C0  RCX: 0x00007FFFF78B8620  RIP: 0x000000000040D855
  R8 : 0x0000000000000004  R9 : 0x0000000000004600  R10: 0x0000000000000065  R11: 0x0000000000635270  R12: 0x00000000000000F9
  R13: 0x0000000000004610  R14: 0x00000000001D02A0  R15: 0x0000000000000001
  CS: 0033  DS: 0000  ES: 0000  FS: 0000  GS: 0000  SS: 002B				
-----------------------------------------------------------------------------------------------------------------------[code]
=> 0x40d855 <create_subimages16+341>:	cmp    QWORD PTR [rbx+0x20],0x0
   0x40d85a <create_subimages16+346>:	mov    QWORD PTR [rsp+0x10],rax
   0x40d85f <create_subimages16+351>:	mov    rax,QWORD PTR [rsp+0x30]
   0x40d864 <create_subimages16+356>:	je     0x40e508 <create_subimages16+3592>
   0x40d86a <create_subimages16+362>:	cmp    BYTE PTR [rax+0x1e],0x3
   0x40d86e <create_subimages16+366>:	je     0x40e570 <create_subimages16+3696>
   0x40d874 <create_subimages16+372>:	mov    ebx,DWORD PTR [rip+0x21ef2e]        # 0x62c7a8 <image_index>
   0x40d87a <create_subimages16+378>:	test   ebx,ebx
-----------------------------------------------------------------------------------------------------------------------------
create_subimages16 (gfs=gfs@entry=0x62eb20, optimize_flags=optimize_flags@entry=0x2, save_uncompressed=0x1) at opttemplate.c:480
480	    if (gfi->local)
gdb$ exploitable
Description: Access violation
Short description: AccessViolation (21/22)
Hash: 4d2cf1395043a02d5dfca2c3e55cb17c.4d2cf1395043a02d5dfca2c3e55cb17c
Exploitability Classification: UNKNOWN
Explanation: The target crashed due to an access violation but there is not enough additional information available to determine exploitability.
```

[subimages.zip](https://github.com/kohler/gifsicle/files/1096366/subimages.zip)

**Comment by kohler** [comment]
Also could not replicate this.

**Comment by strazzere** [comment]
This is also fixed by a0a365136f44e5519f7f486b00a67387f641d0e8 - reproducable before that.

**Comment by kohler** [comment]
By going back to a0a3651^, I was able to find the actual error (which didn't show up on valgrind). Here is the fix: 1638a43201436f796bdf46d33b87468c089e90b8 Thanks!
